plugins/modules/katello_content_view_version.py

```diff
@@ -203,66 +181,58 @@ def main():
             current_lifecycle_environment=dict(),
         ),
         mutually_exclusive=[['current_lifecycle_environment', 'version']],
-        supports_check_mode=True,
     )
 
-    if has_import_error:
-        module.fail_json(msg=import_error_msg)
+    module.task_timeout = 60 * 60
 
-    set_task_timeout(3600000)  # 60 minutes
+    entity_dict = module.clean_params()
 
-    params_dict = dict(
-        [(k, v) for (k, v) in module.params.items() if v is not None])
+    module.connect()
 
-    server_url = module.params['server_url']
-    username = module.params['username']
-    password = module.params['password']
-    verify_ssl = module.params['verify_ssl']
-    state = module.params['state']
+    entity_dict['organization'] = module.find_resource_by_name('organizations', entity_dict['organization'], thin=True)
+    scope = {'organization_id': entity_dict['organization']['id']}
 
-    try:
-        create_server(server_url, (username, password), verify_ssl)
-    except Exception as e:
-        module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+    content_view = module.find_resource_by_name('content_views', name=entity_dict['content_view'], params=scope)
 
-    ping_server(module)
-
-    organization = find_organization(module, params_dict['organization'])
-    content_view = find_content_view(module, name=params_dict['content_view'], organization=organization)
-
-    if 'current_lifecycle_environment' in params_dict:
-        params_dict['current_lifecycle_environment'] = find_lifecycle_environment(module, name=params_dict['current_lifecycle_environment'],
-                                                                                  organization=organization)
-        content_view_version = find_content_view_version(module, content_view, environment=params_dict['current_lifecycle_environment'])
-    elif 'version' in params_dict:
-        content_view_version = find_content_view_version(module, content_view, version=params_dict['version'], failsafe=True)
+    if 'current_lifecycle_environment' in entity_dict:
+        entity_dict['current_lifecycle_environment'] = module.find_resource_by_name(
+            'lifecycle_environments', name=entity_dict['current_lifecycle_environment'], params=scope)
+        search = "content_view_id={},environment_ids=[{}]".format(content_view['id'], entity_dict['current_lifecycle_environment']['id'])
+        content_view_version = module.find_resource('content_view_versions', search=search, thin=True)
+    elif 'version' in entity_dict:
+        search = "content_view_id={},version={}".format(content_view['id'], entity_dict['version'])
+        content_view_version = module.find_resource('content_view_versions', search=search, failsafe=True)
```

Comment:
why failsafe here vs thin in line 201?

Reply:
I guess it could be thin here, too, but the `failsafe` are ok. If we specify lce, and there is no voersion, we want to error out. If we specify a version, and there is none, we ought to create one.
Also we never call update on an existing version. So always thin should be ok.

Reply:
Sounds good. I was basically thinking: "two similar lines, could they be collapsed to one" but your explanation is sound that they can't :)

Reply:
Changed it to thin.